webui/src/main.tsx

```diff
@@ -56,155 +38,128 @@ const mainStyles = (theme: Theme) => createStyles({
     }
 });
 
-class MainComponent extends React.Component<MainComponent.Props, MainComponent.State> {
-
-    constructor(props: MainComponent.Props) {
-        super(props);
+const MainComponent = (props: Main.Props) => {
+    const [user, setUser] = React.useState<UserData | undefined>();
+    const [userLoading, setUserLoading] = React.useState<boolean>(true);
+    const [error, setError] = React.useState<string>('');
+    const [isErrorDialogOpen, setIsErrorDialogOpen] = React.useState<boolean>(false);
```

Comment:
Why do you prefer a function over a class here?

Reply:
Thanks! there was no specific reason for converting it to function. To me, function(hooks) feel cleaner i.e there is no `this` in the API. If you like then I'm happy to convert it back to a class :slightly_smiling_face:

Reply:
Then please keep the previous class-based form. We can switch to functions when we see benefits from such a refactoring, but for now I'd rather keep the code consistent with the rest of webui.